vassal-app/src/main/java/VASSAL/build/module/ChessClockControl.java

```diff
@@ -0,0 +1,843 @@
+  public static final String BUTTON_TOOLTIP = "buttonTooltip"; //$NON-NLS-1$
+
+  public static final String PAUSE_HOTKEY = "pauseHotkey";
+  public static final String NEXT_HOTKEY = "nextHotkey";
+  public static final String SHOW_HOTKEY = "showHotkey";
+
+  public static final String SHOW_TENTHSECONDS = "showTenths";
+  public static final String SHOW_SECONDS = "showSeconds";
+  public static final String SHOW_HOURS = "showHours";
+  public static final String SHOW_DAYS = "showDays";
+
+  public static final String STYLE_ALWAYS = "Always";
+  public static final String STYLE_AUTO   = "Auto";
+  public static final String STYLE_NEVER  = "Never";
+
+  public static final String CHESSMENU_PAUSE = Resources.getString("ChessClock.pause");
+  public static final String CHESSMENU_SHOW  = Resources.getString("ChessClock.show");
+  public static final String CHESSMENU_HIDE  = Resources.getString("ChessClock.hide");
+  public static final String CHESSMENU_NEXT  = Resources.getString("ChessClock.next");
+
+  public static final String COMMAND_PREFIX = "CLOCKCONTROL" + DELIMITER; //NON-NLS-1$
+
+  public ChessClockControl() {
+    setConfigureName("Chess Clock Control");
+
+    chessClocksVisible = false;
+    onlineGame         = false;
+
+    ActionListener al = e -> pressControlButton();
+
+    chessClockButton = new LaunchButton("Chess Clocks", BUTTON_TOOLTIP, BUTTON_TEXT, SHOW_HOTKEY, ICON, al); //$NON-NLS-1$
+    chessClockButton.setToolTipText("Tooltip Text");
+    //chessClockButton.setAttribute(ICON, "chessclock.png"); //BR// I have a decent royalty-free icon, but need to figure out vassal-wide licensing ... or just leave it open for module designer?
+    chessClockButton.addMouseListener(new ChessMouseListener());
+
+    // Set up listeners for hotkeys
+    nextListener = new NamedKeyStrokeListener(e -> {
+      Command c = startNextClock();
+      if ((c != null) && !c.isNull()) {
+        c.execute();
+        GameModule.getGameModule().sendAndLog(c);
+      }
+    });
+    GameModule.getGameModule().addKeyStrokeListener(nextListener);
+
+    showListener = new NamedKeyStrokeListener(e -> pressControlButton());
+    GameModule.getGameModule().addKeyStrokeListener(showListener);
+
+    pauseListener = new NamedKeyStrokeListener(e -> {
+      Command command = stopAllClocks();
+      command.execute();
+      GameModule.getGameModule().sendAndLog(command);
+    });
+    GameModule.getGameModule().addKeyStrokeListener(pauseListener);
+  }
+
+  /**
+   * @return Our list of chessclocks
+   */
+  public Set<ChessClock> getChessClocks() {
+    return chessclocks;
+  }
+
+  /**
+   * Builds the clock control's component hierarchy from a given XML element, or a null one is given initializes
+   * a brand new default "new clock control" hierarchy.
+   * @param e XML element to build from, or null to build the default hierarchy
+   */
+  @Override
+  public void build(Element e) {
+    super.build(e);
+    if (e == null) {
```

Comment:
To reduce indentation levels, consider an early return:
```
if (e != null) {
  return;
}
// rest here
```

Reply:
done